mathics/core/expression.py

```diff
@@ -634,29 +636,70 @@ def evaluate_elements(self, evaluation) -> "Expression":
         return Expression(head, *elements)
 
     def filter(self, head, cond, evaluation):
-        # faster equivalent to: Expression(head, [element in self.leaves if cond(element)])
+        # faster equivalent to: Expression(head, [element in self.elements if cond(element)])
         return structure(head, self, evaluation).filter(self, cond)
 
-    def flatten(
-        self, head, pattern_only=False, callback=None, level=None
+    def flatten_pattern_sequence(self, evaluation):
+        def sequence(element):
+            flattened = element.flatten_pattern_sequence(evaluation)
+            if element.get_head() is SymbolSequence and element.pattern_sequence:
+                return flattened._elements
+            else:
+                return [flattened]
+
+        expr = self._flatten_sequence(sequence, evaluation)
+        if hasattr(self, "options"):
+            expr.options = self.options
+        return expr
+
+    def flatten_sequence(self, evaluation):
+        def sequence(element):
+            if element.get_head_name() == "System`Sequence":
+                return element._elements
+            else:
+                return [element]
+
+        return self._flatten_sequence(sequence, evaluation)
+
+    def flatten_with_respect_to_head(
+        self, head, pattern_only=False, callback=None, level=100
```

Comment:
I don't like this because it draws away from the intent here. 

The purpose of this function is to address an efficiency concern that has been designed into WL that a number of functions or operators can handle lists of operands. And partly due to the way things get parsed or are sometimes expressed we have this kind of nesting of the same operation.  To me it would be like saying that every a lot of time when I add a number add by one so I'll just arbitrarily say that "add" by itself means "add1" . 

What you are noticing with the fact that this happens with list is that there is some sort of parameter list pattern that often comes out with a list where the first argument is also a list.  Better here would be to write some sort of function to deal with that. 

In addition, I think it might be good to add a function like `flatten_head_function`  which drops the head parameter altogether and uses the head as the thing you want to flatten against.  That is what is generally intended.  It is only in the rare situation where a we create a new list in evaluation so the head element is different. It is the name of the function that we are about to replace this with rather than the current head. And I suppose it is done this way for some sort of efficiency. 

All this said, I've run out of time for this week, so let's put these changes on the things to do for next week.